cloudinit/sources/DataSourceAzure.py

```diff
@@ -693,12 +693,22 @@ def exc_cb(msg, exception):
                         self._ephemeral_dhcp_ctx.clean_network()
                         break
 
+                    report_ready_succeeded = self._report_ready(lease=lease)
+                    if not report_ready_succeeded:
+                        msg = ('Failed reporting ready while in '
+                               'the preprovisioning pool.')
+                        report_diagnostic_event(msg)
```

Comment:
Are we intentionally not passing a logger in these `report_diagnostic_event` calls?

Reply:
Awaiting the merging of #563 before I add that here.

Reply:
Merged the code from the branch of #563 to this PR. It now has `logger_func`.